src/nvim/autocmd.c

```diff
@@ -2682,9 +2681,5 @@ void do_filetype_autocmd(buf_T *buf, bool force)
   apply_autocmds(EVENT_FILETYPE, buf->b_p_ft, buf->b_fname, force || ft_recursive == 1, buf);
   ft_recursive--;
 
-  // Just in case the old "buf" is now invalid
-  if (varp != &(buf->b_p_ft)) {
-    varp = NULL;
-  }
```

Comment:
Looks like this is a clobbered remnant of code that used to live in `did_set_string_option` [here](https://github.com/neovim/neovim/blob/ff5cfcdeab2446459ab59d44ce173ef811902c49/src/nvim/optionstr.c#L1550-L1575).

But if the buffer was freed by autocmds, wouldn't using `varp` in a comparison be illegal anyway? :thinking: